inspirehep/bat/pages/holding_panel_literature_list.py

```diff
@@ -60,19 +80,40 @@ def _load_submission_record():
             'Lorem ipsum dolor sit amet, consetetur sadipscing elitr.' in record
         )
 
-    record = _force_load_record(input_data)
-    return ArsenicResponse(_load_submission_record)
+    record = _force_load_record()
+    return ArsenicResponse(_load_record)
 
 
-def _force_load_record(input_data):
-    try:
-        WebDriverWait(Arsenic(), 10).until(
-            EC.visibility_of_element_located(
-                (By.XPATH, '//div[@class="row hp-item ng-scope"][1]/div/div/div[2]/holding-pen-template-handler/div[3]/a'))).click()
-        record = WebDriverWait(Arsenic(), 10).until(
-            EC.visibility_of_element_located((By.XPATH, '//div[@class="row hp-item ng-scope"][1]'))).text
-    except (ElementNotVisibleException, WebDriverException):
+def load_completed_record():
+    def _referesh_page():
         go_to()
-        record = _force_load_record(input_data)
+        return _force_load_record()
+
+    def _force_load_record():
+        try:
+            # click the abstract link
+            WebDriverWait(Arsenic(), 10).until(
+                EC.visibility_of_element_located(
+                    (By.XPATH, '//div[@class="row hp-item ng-scope"][1]/div/div/div[2]/holding-pen-template-handler/div[4]/a')
+                )
+            ).click()
+            # take the record
+            record = WebDriverWait(Arsenic(), 10).until(
+                EC.visibility_of_element_located(
+                    (By.XPATH, '//div[@class="row hp-item ng-scope"][1]')
+                )
+            ).text
+        except (ElementNotVisibleException, WebDriverException):
+            _referesh_page()
+
+        if 'Waiting' in record:
+            return _referesh_page()
```

Comment:
This is necessary because the workflow sometimes needs more time to set the record's status to `Completed`. For this reason if there is the Waiting keyword it means that the workflows requires more time to complete the record process.